tools/cldr-code/src/main/java/org/unicode/cldr/util/UnitConverter.java

```diff
@@ -241,23 +241,113 @@ public UnitConverter cloneAsThawed() {
     public static final class ConversionInfo implements Comparable<ConversionInfo> {
         public final Rational factor;
         public final Rational offset;
+        public String special;
+        public boolean inverse; // only used with special
 
         static final ConversionInfo IDENTITY = new ConversionInfo(Rational.ONE, Rational.ZERO);
 
         public ConversionInfo(Rational factor, Rational offset) {
             this.factor = factor;
             this.offset = offset;
+            this.special = null;
+            this.inverse = false;
+        }
+
+        public ConversionInfo(String special, boolean inverse) {
+            this.factor = Rational.ZERO; // if ONE it will be treated as a base unit
+            this.offset = Rational.ZERO;
+            this.special = special;
+            this.inverse = inverse;
         }
 
         public Rational convert(Rational source) {
+            if (special != null) {
+                if (special.equals("beaufort")) {
+                    return (inverse) ? baseToBeaufort(source) : beaufortToBase(source);
+                }
+                return source;
+            }
             return source.multiply(factor).add(offset);
         }
 
         public Rational convertBackwards(Rational source) {
+            if (special != null) {
+                if (special.equals("beaufort")) {
+                    return (inverse) ? beaufortToBase(source) : baseToBeaufort(source);
+                }
+                return source;
+            }
             return source.subtract(offset).divide(factor);
         }
 
+        private static final BigInteger BEAUFORT_MAX = BigInteger.valueOf(17);
+        private static final Rational[] beaufortToMetersPerSec = {
```

Comment:
These values are *very* close to the midpoint of each bucket, at 12 Bft there's only a difference of 0.1 which could be a rounding error. You could calculate these from the other table.

Reply:
Thanks, I should have checked that. Of course for Bft 17 there is no midpoint... But we can special-case that.

Reply:
I did end up using this suggestion, computing the midpoints from the ranges, thanks.